sqlalchemylogger/handlers.py

```diff
@@ -1,42 +1,109 @@
 import logging
 import traceback
+import sqlalchemy
 from sqlalchemy import create_engine
 from sqlalchemy.orm import sessionmaker
-from .models import Log, Base
+from .models import createLogClass, Base
 from sqlalchemy.exc import OperationalError, InvalidRequestError
+from sqlalchemy_utils import database_exists, create_database
+import threading, queue, time
+from .filters import ContainsExpression, DoesNotContainExpression
+
+
+module_logs = logging.getLogger(__name__)
 
 class SQLAlchemyHandler(logging.Handler):
-    def __init__(self, sqlalchemyUrl):
+    
+    MAX_NB_LOGS = 100
+    MAX_TIMEOUT = 1
+
+    def __init__(self, sqlalchemyUrl, 
+              doesNotContainExpression = None,
+              containsExpression = None):
         super().__init__()
-        self.engine = create_engine(sqlalchemyUrl)
+        # initialize DB session
+        self.engine = create_engine(sqlalchemyUrl['url'])
+        self.Log = createLogClass(
+                  tablename = sqlalchemyUrl.get('tablename', 'logs'),
+                  tableargs = sqlalchemyUrl.get('tableargs', None))
         Base.metadata.bind = self.engine
         DBSession = sessionmaker(bind=self.engine)
         self.session = DBSession()
+        # initialize log queue
+        self.log_queue = queue.Queue()
+        # initialize a thread to process the logs Asynchronously
+        self.processor_thread = threading.Thread(target = self._processor, daemon = True)
+        self.processor_thread.start()
+        # initialize filters
+        if doesNotContainExpression:
+             self.addFilter(DoesNotContainExpression(doesNotContainExpression))
+        if containsExpression:
+             self.addFilter(ContainsExpression(containsExpression))
+
+
+    def _processor(self):
+        _terminated = False
+        while not _terminated:
+            logs = []
+            time_since_last = time.time()
+            while True:
+                try:
+                    log = self.log_queue.get(timeout=self.MAX_TIMEOUT)
+                    if log is None:
+                        # one way of killing a thread
+                        _terminated = True
```

Comment:
I don't get why you stop the thread when there is no log after 1s?

Reply:
`.get()` does not return a `None`, it returns an exception `Empty`. The way to terminate the thread is to `queue.put(None)` excplicitely

Reply:
so the thread is kept alive normally and simply tries to submit logs to `writelogs` each MAX_TIMEOUT